spec/amber/router/params_spec.cr

```diff
@@ -94,16 +80,35 @@ module Amber::Router
     end
 
     describe "#to_h" do
-      headers = HTTP::Headers.new
-      headers["Content-Type"] = "multipart/form-data; boundary=fhhRFLCazlkA0dX; charset=UTF-8"
-      multipart_content = ::File.read(::File.expand_path("spec/support/sample/multipart.txt"))
-      multipart_body = multipart_content.gsub("\n", "\r\n")
-      request = HTTP::Request.new("GET", "/?test=test&test=test2&#{HTTP::Request::METHOD}=put&status=1234", headers, multipart_body)
-      params = Params.new(request)
-
       it "returns a hash with all params" do
+        params = multipart_form_post
         params.to_h.keys.should eq %w(test _method status _csrf title content)
+        params.to_h["test"].should eq "test"
+      end
+
+      it "returns the first value for key" do
+        url_encoded_form_post.to_h["cat"].should eq "1"
       end
     end
   end
 end
+
+def url_encoded_form_post
```

Comment:
These methods need to be a part of a module above and marked as `protected`/`private`. Otherwise, they can be accessed in other files.